src/components/MagicCodeInput.js

```diff
@@ -224,8 +210,7 @@ function MagicCodeInput(props) {
         let numbers = decomposeString(props.value, props.maxLength);
         numbers = [...numbers.slice(0, editIndex), ...numbersArr, ...numbers.slice(numbersArr.length + editIndex, props.maxLength)];
 
-        setFocusedIndex(updatedFocusedIndex);
-        setInput(value);
+        inputRefs.current[updatedFocusedIndex].focus();
```

Comment:
We used to set the input to `value` but with this change we won't, the focus method will set the input to an empty string. This does not seem right. Can you please fix this

Reply:
I intentionally take out the value bcoz I think we need to reset it once the value has been given for the input box. Its impact can be seen in the last input box where the focusedIndex won't be changed if the full code isn't Fulfilled.

https://github.com/Expensify/App/assets/14358475/c549186a-7df1-44d1-bc88-81dada28d199